tests/unit/restapi/v1/test_queue.py

```diff
@@ -0,0 +1,500 @@
+    )
+    assert response.status_code == 200 and response.get_json()["name"] == expected_name
+
+
+def assert_queue_is_not_found(
+    client: FlaskClient,
+    queue_id: int,
+) -> None:
+    """Assert that a queue is not found.
+
+    Args:
+        client: The Flask test client.
+        queue_id: The id of the queue to retrieve.
+
+    Raises:
+        AssertionError: If the response status code is not 404.
+    """
+    response = client.get(
+        f"/{V1_ROOT}/{V1_QUEUES_ROUTE}/{queue_id}",
+        follow_redirects=True,
+    )
+    assert response.status_code == 404
+
+
+def assert_cannot_rename_queue_with_existing_name(
+    client: FlaskClient,
+    queue_id: int,
+    existing_name: str,
+    existing_description: str,
+) -> None:
+    """Assert that renaming a queue with an existing name fails.
+    Args:
+        client: The Flask test client.
+        queue_id: The id of the queue to rename.
+        name: The name of an existing queue.
+    Raises:
+        AssertionError: If the response status code is not 400.
+    """
+    response = modify_queue(
+        client=client,
+        queue_id=queue_id,
+        new_name=existing_name,
+        new_description=existing_description,
+    )
+    assert response.status_code == 400
+
+
+# -- Tests -----------------------------------------------------------------------------
+
+
+@pytest.mark.v1
+def test_create_queue(
+    client: FlaskClient,
+    db: SQLAlchemy,
+    auth_account: dict[str, Any],
+) -> None:
+    """Test that queues can be correctly registered and retrieved using the API.
+
+    Given an authenticated user, this test validates the following sequence of actions:
+
+    - The user registers a queue named "tensorflow_cpu".
+    - The response is valid matches the expected values given the registration request.
+    - The user is able to retrieve information about the queue using the queue id.
+    """
+    name = "tensorflow_cpu"
+    description = "The first queue."
+    user_id = auth_account["user_id"]
+    group_id = auth_account["default_group_id"]
+    queue1_response = actions.register_queue(
+        client, name=name, description=description, group_id=group_id
+    )
+    queue1_expected = queue1_response.get_json()
+    assert_queue_response_contents_matches_expectations(
+        response=queue1_expected,
+        expected_contents={
+            "name": name,
+            "description": description,
+            "user_id": user_id,
+            "group_id": group_id,
+        },
+    )
+    assert_retrieving_queue_by_id_works(
+        client, queue_id=queue1_expected["queueId"], expected=queue1_expected
+    )
+
+
+@pytest.mark.v1
+def test_queue_get_all(
+    client: FlaskClient,
+    db: SQLAlchemy,
+    auth_account: dict[str, Any],
+    registered_queues: dict[str, Any],
+) -> None:
+    """Test that all queues can be retrieved.
+
+    Given an authenticated user and registered queues, this test validates the following
+        sequence of actions:
```

Comment:
Minor whitespace nitpick, the prose paragraphs at the top of a Google-style docstring don't need to be indented on subsequent lines:

```suggestion
    Given an authenticated user and registered queues, this test validates the following
    sequence of actions:
```

This comment also applies to other docstrings in the file.